src/components/editor/EditorLogin.jsx

```diff
@@ -51,6 +85,29 @@ export default class Login extends BaseComponent {
     }
   }
 
+  onGoogleResponse(response) {
+    if (response.error) {
+      alert(`Google Login Failed.\n${response.error}`);
+      return;
+    }
+
+    const token = response.tokenId;
+
+    const xhr = new XMLHttpRequest();
+    xhr.open('POST', '/googlelogin', false);
+    xhr.setRequestHeader('Content-type', 'application/x-www-form-urlencoded');
+
+    xhr.onreadystatechange = () => {
+      if (xhr.readyState === XMLHttpRequest.DONE && xhr.status === 200) {
+        browserHistory.push('/');
+      } else {
+        alert(`Google Login Failed.\n${xhr.status}: ${xhr.statusText}`);
+      }
+    };
+
+    xhr.send(`data=${encodeURIComponent(token)}`);
```

Comment:
Hmm... This could simply be because you actually have more experience than me with writing these requests, so bear with me, but is there any reason you're doing URL encoding on this? Couldn't you just send it in plain text, or is it because you on the serverside send it sa part of a GET request? Because in that case you of course need to URI encode it.

I guess this could be best practice, so you're manually writing out a form-urlencoding here yeah? I just think writing just plain text as it's just a single message or maybe JSON could be just as easy and a bit more modern, no? Thinking it may be a bit overkill

Reply:
Emm I'm actually not super familiar with it either, I'll try to send as plain text and see. It should work that way too, I hopefully just need to change the header to `text/plain`

Reply:
Okay, in that case I definitely think I'll say you should just use `text/plain` and send it as a plain string, no reason not to. And in the future if you have several things to send, I would probably also just send it as JSON actually, I think that's the more modern way vs form encoding.

Beware that you should probably URL encode the token before you send it in the GET request to the Google servers though

Reply:
okay plain text works